src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -33,14 +33,16 @@ internal class AzureAppConfigurationProvider : ConfigurationProvider, IConfigura
         private ConcurrentDictionary<KeyValueIdentifier, ConfigurationSetting> _watchedSettings = new ConcurrentDictionary<KeyValueIdentifier, ConfigurationSetting>();
 
         private readonly TimeSpan MinCacheExpirationInterval;
-        private readonly SemaphoreSlim InitializationSemaphore = new SemaphoreSlim(1);
 
         // The most-recent time when the refresh operation attempted to load the initial configuration
         private DateTimeOffset InitializationCacheExpires = default;
 
         private static readonly TimeSpan MinDelayForUnhandledFailure = TimeSpan.FromSeconds(5);
         private static readonly TimeSpan DefaultMaxSetDirtyDelay = TimeSpan.FromSeconds(30);
 
+        // To avoid concurrent network operations, this flag is used to achieve synchronization between multiple threads.

```

Comment:
>// To avoid concurrent [](http://example.com/codeflow?start=8&length=22)

I would say watch out for comments that might be implied by the variable name alone. Otherwise we may end up with a lot of comment bloat.

Reply:
I agree. In this case, since the use of this variable is more complicated, I wanted to explain the concept too for future readers.